tc/core/polyhedral/memory_promotion.cc

```diff
@@ -135,13 +135,12 @@ std::unique_ptr<TensorReferenceGroup> TensorReferenceGroup::makeSingleton(
   return group;
 }
 
-isl::set ScopedFootprint::footprint(isl::set domain) const {
-  auto space = add_range(domain.get_space(), size());
-  auto accessed = isl::map::universe(space).intersect_domain(domain);
+isl::set TensorReferenceGroup::approximateFootprint() const {
+  auto accessed = scopedAccesses();
```

Comment:
I don't understand this code.
You apparently start out from the actual accesses and then you "intersect"
that with an overapproximation of those accesses.
Doesn't that just give you the actual accesses again?

Reply:
good point